tests/isTextSame.py

```diff
@@ -1,47 +1,65 @@
 #!/usr/bin/python3
 #---Dependencies---
-#sudo apt install poppler util
-
+#poppler-utils
 import subprocess
+import os
 
-#Run pdftotext command in linux bash
-subprocess.call(['pdftotext', '../file.pdf', 'output'])
-
-#grab asciidoc
-adoc = open('../file.adoc', 'r')
-adocTxt = adoc.read()
-
-#get rid of + in asciidoc
-adocTxt = adocTxt.replace('+', '')
-#split string into each line
-adocLines = adocTxt.split('\n')
-    
-adocTxt = ''
-#add lines back together removing weird ones
-for line in adocLines:
-    if not line.startswith('[#'):
-        if not line.startswith('Page: link:#'):
-            adocTxt += line
-#Remove line breaks
-adocTxt = adocTxt.replace('\n', '')
-#remove spaces
-adocTxt = adocTxt.replace(' ', '')
-
-#grab output file
-pdfOutput = open('output', 'r')
-pdfTxt = pdfOutput.read()
+#pathname variables
+PATH_TO_PDF = '../conversions/pdfs/'
+PATH_TO_ADOC = '../conversions/adocs/'
+total = 0
+passed = 0
 
-#remove line breaks
-pdfTxt = pdfTxt.replace('\n', '')
-pdfTxt = pdfTxt.replace(' ', '')
+def format_adoc_txt(self):
+    #get rid of + in asciidoc
+    self = self.replace('+', '')
+    #split string into each line
+    adoc_lines = self.split('\n')
+    self = ''
+    #add lines back together removing weird ones
+    for line in adoc_lines:
+        if not line.startswith('[#'):
+            if not line.startswith('Page: link:#'):
+                self += line
+    #Remove line breaks
+    self = self.replace('\n', '')
+    #remove spaces
+    self = self.replace(' ', '')
+    return self
 
-if adocTxt  in pdfTxt:
-    print("Pass")
-else:
-    print("Fail")
-    diff = set(char for char in adocTxt) - set(char for char in pdfTxt)
-    print('Differing characters: ' + diff)
+def format_pdf_txt(self):
+    #remove line breaks
+    self = self.replace('\n', '')
+    self = self.replace(' ', '')
+    return self
 
-adoc.close()
-pdfOutput.close()
-subprocess.call(['rm', 'output'])
+for pdf in os.listdir(PATH_TO_PDF):
+    #set pdf file name variable
+    pdf_name = os.fsdecode(pdf)
+    #make sure current or previous directory isn't selected
+    if pdf_name == '.' or pdf_name == '..':
+        continue
+    total += 1
+    #call pdftotext command in bash
+    subprocess.call(['pdftotext', PATH_TO_PDF + pdf_name, pdf_name + '.output'])
```

Comment:
Can an exception be thrown from `subprocess.call()`?

Reply:
If using Python 3.5+, can utilize new and improved`subprocess.run()` method instead of `subprocess.call()`

https://docs.python.org/3/library/subprocess.html#older-high-level-api